requests_pkcs12.py

```diff
@@ -165,20 +174,46 @@ def selftest():
         cryptography.hazmat.primitives.hashes.SHA512(),
         cryptography.hazmat.backends.default_backend()
     )
-    pkcs12_data = cryptography.hazmat.primitives.serialization.pkcs12.serialize_key_and_certificates(
-        name=b'test',
-        key=key,
-        cert=cert,
-        cas=[cert, cert, cert],
-        encryption_algorithm=cryptography.hazmat.primitives.serialization.BestAvailableEncryption(b'correcthorsebatterystaple')
-    )
-    response = get(
-        'https://example.com/',
-        pkcs12_data=pkcs12_data,
-        pkcs12_password='correcthorsebatterystaple'
-    )
-    if response.status_code != 200:
-        raise Exception('Unexpected response: {response!r}'.format(**locals()))
+
+    test_cases = {
```

Comment:
This structure defines a DSL that describes what we can more explicitly achieve by function calls. So there should be some function like "execute_test_case" (feel free to choose a better name) that is just called multiple times. See also the following comments.